umbraco/Umbraco-CMS
# V15: Update to dotnet 9

# Notes
- Had to rename `TrimStart` & `TrimEnd`, as they are now implemented by Microsoft, and the behavior differs from ours.
- Updates dotnet verison to dotnet9
- Updates nuget packages to their respective dotnet 9 version
- Had to update Umbraco.Code to 2.2 aswell, as it used the `Microsoft.CodeAnalysis` package, and thus we couldn't update that.

# How to test
Ensure both templates & source code work with dotnet 9
- Download dotnet 9 preview here: https://dotnet.microsoft.com/en-us/download/dotnet/9.0
- In the root of the project, run `dotnet pack -o nuget`
- Go to the nuget folder `cd nuget`
- Copy the path of that folder, and then run `dotnet nuget add source [PATH TO NUGET FOLDER HERE]`
- Install the templates, take note of the version of the templates nupkg in the nuget folder `dotnet new install Umbraco.Templates::[version of templates here]`
- Create a new umbraco project with the new template `dotnet new umbraco -n "MyTestProject"`
- Run the project and assert it works 👍
